### PR TITLE
Update 'Post' model

### LINKED ISSUE #71: Updating Post model to contain a Profile/User (preferably user) field instead of nameOfPoster
Do we want to allow users with no profiles? if the answer is nom then this issue is needed, otherwise for now we can update the Post model to be linked with the Django default User (will be conflicted with the PR related to issue #55)

### PR DESCRIPTION
**Pull request by Shoval-Ovadia**
The update contains a change of the 'nameOfPoster' field to a user field.
This update resulted in changes to the following files:
- 'test data' files
- test files
- HTML files

Detailed explanations in the commits messages.
 
closes #71

### CODE REVIEW

**Comment by maayanhd** [comment]
It's working properly without the test data of post, so something there is causing this issue

**Comment by Shoval-Ovadia** [comment]
> It's working properly without the test data of post, so something there is causing this issue, as we can see it's triggered. Also this seems to work (added dependency in the Post test_data file in the user initial migration file, as we did at the time with the commenting test_data -
> 
> from django.db import migrations, transaction
> from django.contrib.auth.models import User
> 
> class Migration(migrations.Migration):
> dependencies = [
> ('users', '0001_initial'),
> ('Post', '0001_initial'),
> ]
> 
> ```
> def generate_data(apps, schema_editor):
>     from Post.models import Post
> 
>     p1_Author = User.objects.create_user(
>         'Shoval', 'Test10@gmail.com', 'password777'
>     )
>     p1_Place = 'Eilat'
>     p1_URL = (
>         "https://israel.travel/wp-content/uploads/2019/02/eilatnewinside-min.jpg"
>     )
>     p1_Description = 'This is my favorite place! chill vibes and beautiful sea.'
> 
>     p2_Author = User.objects.create_user(
>         'Daniel', 'Test20@gmail.com', 'password777'
>     )
>     p2_Place = 'Dead Sea'
>     p2_URL = "https://velvetescape.com/wp-content/uploads/2011/11/IMG_2370-3-1280x920.jpg"
>     p2_Description = 'Beautiful place.'
> 
>     with transaction.atomic():
>         Post(
>             user=p1_Author,
>             nameOfLocation=p1_Place,
>             photoURL=p1_URL,
>             Description=p1_Description,
>         ).save()
>         Post(
>             user=p2_Author,
>             nameOfLocation=p2_Place,
>             photoURL=p2_URL,
>             Description=p2_Description,
>         ).save()
> 
> operations = [migrations.RunPython(generate_data)]
> ```

I added what you asked for.
Still, there is a problem with the tests.

**Comment by Yarboa** [comment]
> > It's working properly without the test data of post, so something there is causing this issue, as we can see it's triggered. Also this seems to work (added dependency in the Post test_data file in the user initial migration file, as we did at the time with the commenting test_data -
> > from django.db import migrations, transaction
> > from django.contrib.auth.models import User
> > class Migration(migrations.Migration):
> > dependencies = [
> > ('users', '0001_initial'),
> > ('Post', '0001_initial'),
> > ]
> > ```
> > def generate_data(apps, schema_editor):
> >     from Post.models import Post
> > 
> >     p1_Author = User.objects.create_user(
> >         'Shoval', 'Test10@gmail.com', 'password777'
> >     )
> >     p1_Place = 'Eilat'
> >     p1_URL = (
> >         "https://israel.travel/wp-content/uploads/2019/02/eilatnewinside-min.jpg"
> >     )
> >     p1_Description = 'This is my favorite place! chill vibes and beautiful sea.'
> > 
> >     p2_Author = User.objects.create_user(
> >         'Daniel', 'Test20@gmail.com', 'password777'
> >     )
> >     p2_Place = 'Dead Sea'
> >     p2_URL = "https://velvetescape.com/wp-content/uploads/2011/11/IMG_2370-3-1280x920.jpg"
> >     p2_Description = 'Beautiful place.'
> > 
> >     with transaction.atomic():
> >         Post(
> >             user=p1_Author,
> >             nameOfLocation=p1_Place,
> >             photoURL=p1_URL,
> >             Description=p1_Description,
> >         ).save()
> >         Post(
> >             user=p2_Author,
> >             nameOfLocation=p2_Place,
> >             photoURL=p2_URL,
> >             Description=p2_Description,
> >         ).save()
> > 
> > operations = [migrations.RunPython(generate_data)]
> > ```
> 
> I added what you asked for.
> Still, there is a problem with the tests.

Did you try to work similar to this ? 
commenting_system/migrations/0002_test_data.py

I think you have an issue that user is not saved()

**Comment by Yarboa** [comment]
Based on pur slack chat

Post/migrations/0002_test_data.py
-        ('users', '0001_initial'),
+        ('users', '__latest__'),

 users/migrations/0002_test_data.py
+        ('auth', '__latest__'),
         ('users', '0001_initial'),

Will do the trick

File: Post/migrations/0002_test_data.py [hunk 1]
```diff
@@ -1,35 +1,42 @@
...
 
 
 class Migration(migrations.Migration):
     dependencies = [
+        ('users', '0001_initial'),
+        ('users', '__latest__'),
         ('Post', '0001_initial'),
     ]
 
     def generate_data(apps, schema_editor):
```
Comment (Yarboa) [inline Post/migrations/0002_test_data.py:12]: please refer commenting test_data. i would advise implement it the same.
user_list, posts_list and for loop under with transaction

Comment (Shoval-Ovadia) [reply Post/migrations/0002_test_data.py:11]: I changed the file as you asked.

**Review by Leeadar — CHANGES_REQUESTED** [review]
I got this error:

image

**Comment by Shoval-Ovadia** [comment]
> I got this error:
> 
> image

Please try the following:

- vagrant destroy
- delete the DB file
- vagrant up --provision

waiting for your update.

**Comment by Yarboa** [comment]
> > I got this error:
> > image
> 
> Please try the following:
> 
> * vagrant destroy
> * delete the DB file
> * vagrant up --provision
> 
> waiting for your update.

Work for me, also

**Review by Leeadar — APPROVED** [review]
Works now
LGTM

**Review by Yarboa — APPROVED** [review]

**Pull request merged**